Changes in sourcegen break langchain4j module

branch `2.0.x`

```
* What went wrong:
Execution failed for task ':micronaut-langchain4j-processor:compileJava'.
> Compilation failed; see the compiler output below.
  /Users/sdelamo/github/micronaut-projects/micronaut-langchain4j/micronaut-langchain4j-processor/src/main/java/io/micronaut/langchain4j/processor/Langchain4jConfigVisitor.java:356: warning: non-varargs call of varargs method with inexact argument type for last parameter;
                          .addMember("excludes", Arrays.asList(allExcludes))
                                                               ^
    cast to Object for a varargs call
    cast to Object[] for a non-varargs call and to suppress this warning
  1 error
```

--- a/core/src/main/java/io/micronaut/core/util/ArrayUtils.java
+++ b/core/src/main/java/io/micronaut/core/util/ArrayUtils.java
@@ -95,7 +95,6 @@ public final class ArrayUtils {
      * @param <T> The array type
      * @return The concatenated array
      */
-    @SafeVarargs
     public static <T> T[] concat(T[] a, T... b) {
         int bLen = b.length;
 

